src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -287,6 +308,66 @@ public TestResult findCorrespondingResult(String id) {
         return Collections.emptyList();
     }
 
+    /**
+     * Start for Test Result Trends.
+     * From here onwards start the code area responsible for generating various test result trends.
+     */
+
+    public hudson.tasks.junit.TestResult loadXml(){return null;}
+
+    /**
+     * A method for getting the list of packages for all levels of hierarchy.
+     * @return Array of packages of all hierarchies.
+     */
+    public String[] getProjectList(){
+
+        /*
+         * If project list is already ready no need to construct again.
+         */
+        if(projectList!=null) return projectList;
```

Comment:
Indentation. Both eclipse and Intellij have shortcuts you can directly use.

Reply:
So, should I write return on next line.

Reply:
Yes, also `if (projectList!=null)` always put space in b/w. Use your IDE indentation format, it would take care of these.